pkg/blob/controllerserver.go

```diff
@@ -466,16 +442,11 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Errorf(codes.Internal, "failed to create container(%s) on account(%s) type(%s) rg(%s) location(%s) size(%d), error: %v", validContainerName, accountName, storageAccountType, resourceGroup, location, requestGiB, err)
 	}
 	if volContentSource != nil {
-		dstAzcopyAuthEnv := srcAzcopyAuthEnv
-		dstAccountSASToken := srcAccountSASToken
-		if srcAccountName != accountName {
-			if dstAccountSASToken, dstAzcopyAuthEnv, err = d.getAzcopyAuth(ctx, accountName, accountKey, storageEndpointSuffix, accountOptions, secrets, secretName, secretNamespace); err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to getAzcopyAuth on account(%s) rg(%s), error: %v", accountOptions.Name, accountOptions.ResourceGroup, err)
-			}
+		accountSASToken, authAzcopyEnv, err := d.getAzcopyAuth(ctx, accountName, accountKey, storageEndpointSuffix, accountOptions, secrets, secretName, secretNamespace)
```

Comment:
If src `getAzcopyAuth` fails after dst blob container is created the operation will be retried. 

Will the controller server then create another new blob container? and another?

Checking volume source auth after creating the destination container could lead to an unbounded number of blob containers being created.

Reply:
CSI will retry to create this volume again. And since the VolumeID is the same, controller server will create the same container.